### PR TITLE
0.15 Editorial Pass 2

### PR DESCRIPTION
**Pull request by cart**
* Pares down the picking example to its bare essentials
* Adapts property animation to use the changes in https://github.com/bevyengine/bevy/pull/16484
* Adds some clarity to the animation mask section

### CODE REVIEW

**Review by djeedai — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
We shouldn't merge this until the linked animation PR lands, to make sure we're actually showing the final API.

File: release-content/0.15/release-notes/15800_Add_mesh_picking_backend_and_MeshRayCast_system_parameter.md
Comment (richchurcher) [inline release-content/0.15/release-notes/15800_Add_mesh_picking_backend_and_MeshRayCast_system_parameter.md:25]:
```suggestion
- Mesh: this is a naive raycast against the full mesh. If you run into performance problems here, you should use simplified meshes and an acceleration data structure like a [BVH](https://en.wikipedia.org/wiki/Bounding_volume_hierarchy) to speed this up. As a result, this functionality is currently disabled by default. It can be enabled by enabling the [`MeshPickingPlugin`].
```

**Comment by cart** [comment]
Just accounted for changes in https://github.com/bevyengine/bevy/pull/16484. This should be ready to merge once that lands

**Pull request merged**
